Twitch EventSub library/SubsRegister/Register.cs

```diff
@@ -0,0 +1,731 @@
+            Ver = "1",
+            SpecificObject = typeof(StreamOfflineEvent),
+            SubscriptionType = SubscriptionType.StreamOffline,
+            Conditions = CondList(ConditionType.BroadcasterUserId)
+        };
+
+        public static readonly RegisterItem RegStreamOnline = new RegisterItem
+        {
+            Key = RegisterKeys.StreamOnline,
+            Ver = "1",
+            SpecificObject = typeof(StreamOnlineEvent),
+            SubscriptionType = SubscriptionType.StreamOnline,
+            Conditions = CondList(ConditionType.BroadcasterUserId)
+        };
+
+        public static readonly RegisterItem RegChannelSuspiciousUserMessage = new RegisterItem
+        {
+            Key = RegisterKeys.ChannelSuspiciousUserMessage,
+            Ver = "1",
+            SpecificObject = typeof(ChannelSuspiciousUserMessageEvent),
+            SubscriptionType = SubscriptionType.SuspiciousUserMessage,
+            Conditions = CondList(ConditionType.BroadcasterUserId, ConditionType.ModeratorUserId)
+        };
+
+        public static readonly RegisterItem RegChannelSuspiciousUserUpdate = new RegisterItem
+        {
+            Key = RegisterKeys.ChannelSuspiciousUserUpdate,
+            Ver = "1",
+            SpecificObject = typeof(ChannelSuspiciousUserUpdateEvent),
+            SubscriptionType = SubscriptionType.SuspiciousUserUpdate,
+            Conditions = CondList(ConditionType.BroadcasterUserId, ConditionType.ModeratorUserId)
+        };
+
+        public static readonly List<RegisterItem> RegisterList = GetRegisterList();
+
+        public static readonly Dictionary<string, RegisterItem> RegisterDictionary = GetRegisterDictionary();
+
+        public static List<RegisterItem> GetRegisterList()
+        {
+            var registryKeysList = RegisterKeys.KeysList;
+            var registryItems = new List<RegisterItem>();
+
+            foreach (var key in registryKeysList)
+            {
+                var item = GetRegistryItem(key);
+                if (item != null)
+                {
+                    registryItems.Add(item);
+                }
+            }
+
+            return registryItems;
+        }
+
+        public static Dictionary<string, RegisterItem> GetRegisterDictionary()
+        {
+            var registryKeysList = RegisterKeys.KeysList;
+            var registryItems = new Dictionary<string, RegisterItem>();
+
+            foreach (string key in registryKeysList)
+            {
+                var item = GetRegistryItem(key);
+                if (item != null)
+                {
+                    registryItems.Add(key, item);
+                }
+            }
+
+            return registryItems;
+        }
+
+        // Retrieves the RegistryItem based on the event key
+        public static RegisterItem GetRegistryItem(string key)
+        {
+            return key switch
+            {
+                RegisterKeys.AutomodMessageHold => RegAutomodMessageHold,
```

Comment:
Just wondering, why you don't use `RegisterDictionary` instead of manual mapping? Seems like hassle to deal with all those types.

Reply:
Dictionary cant be used, since GetRegistryItem fills list and list fills dictionary. 
In theory I could use another refrection and use it to build it , but since its all at one place, the change will not be that big of a deal. I will add it to TODOS.

The main idea of this change was to get rid of multiple places where changes happedened and get rid of repeated copy paste mistakes.